### PR TITLE
Command security

### PR DESCRIPTION
**Pull request by lrvick**
- drop pty
- drop port forwarding
- force all commands to go through git-shell
- force all tests through "login" command

### CODE REVIEW

File: test/test.bats [hunk 1]
```diff
@@ -370,10 +375,7 @@ ${lines[1]}
 	 run ssh_command "$command"
 	 [ $status -eq 0 ]
 
-	ssh -v -p2222 \
-		-a -i test-keys/test-sshkey2 \
-		-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
-		git@git-deploy
+    run ssh_command "user" test-keys/test-sshkey2
```
Comment (jlai) [inline test/test.bats]: wrong indent?

**Comment by lrvick** [comment]
Travis works thanks to some debugging help from @jlai jason and a workaround for some flaky docker behaviour when using bash: https://github.com/docker/compose/issues/3379

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,7 @@
 FROM debian:jessie
 
-RUN apt-get update && \
+RUN sed -i "s/httpredir.debian/mirrors.kernel/g" /etc/apt/sources.list && \
```
Comment (thepwagner) [inline Dockerfile]: For science! (grabbing a random fat .deb):
```
pwagner@pebble ~ on master ✗  $ curl -o /dev/null -L http://httpredir.debian.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  334M  100  334M    0     0  16.6M      0  0:00:20  0:00:20 --:--:-- 18.0M
pwagner@pebble ~ on master ✗  $ 
pwagner@pebble ~ on master ✗  $ curl -o /dev/null -L http://httpredir.debian.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  334M  100  334M    0     0  41.2M      0  0:00:08  0:00:08 --:--:-- 43.4M
pwagner@pebble ~ on master ✗  $ curl -o /dev/null -L http://httpredir.debian.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  334M  100  334M    0     0  45.8M      0  0:00:07  0:00:07 --:--:-- 45.8M


pwagner@pebble ~ on master ✗  $ curl -o /dev/null http://mirrors.kernel.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  334M  100  334M    0     0  30.7M      0  0:00:10  0:00:10 --:--:-- 30.8M
pwagner@pebble ~ on master ✗  $ curl -o /dev/null http://mirrors.kernel.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  334M  100  334M    0     0  18.9M      0  0:00:17  0:00:17 --:--:-- 18.8M
pwagner@pebble ~ on master ✗  $ curl -o /dev/null http://mirrors.kernel.org/debian/pool/main/l/linux/linux-image-3.16.0-4-amd64-dbg_3.16.7-ckt25-2_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  334M  100  334M    0     0  29.3M      0  0:00:11  0:00:11 --:--:-- 27.2M
```

Quick tests say this is slower from us-west-2 (and more SPOF)?

File: bin/user [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+echo $CURRENT_USER
```
Comment (thepwagner) [inline bin/user]: +1! Why didn't we have this before :P

File: test/test.bats [hunk 4]
```diff
@@ -370,10 +375,7 @@ ${lines[1]}
...
 	[ $status -eq 0 ]
 }
```
Comment (thepwagner) [inline test/test.bats]: Test addition: attempt a nasty command (`rm -Rf ~`?) and verify it's rejected?

**Comment by thepwagner** [comment]
+1; can't wait to fiddle with pipes on top of this (WEBOPS-693).

File: Dockerfile (hunk 2 above)
Comment (lrvick) [inline Dockerfile]: Just found this though: https://github.com/docker-library/buildpack-deps/issues/40

Looks like all my failures might be to dirty apt cache in the way they build images. Putting apt-get clean in instead to see if this resolves things for us.

File: Dockerfile [hunk 5]
```diff
@@ -1,6 +1,7 @@
 FROM debian:jessie
 
-RUN apt-get update && \
+RUN aot-get clean && \
```
Comment (DeviaVir) [inline Dockerfile]: `aot`?

Comment (lrvick) [inline Dockerfile]: that would do it

**Pull request merged**